contracts/ERC20TimedMint.sol

```diff
@@ -1,23 +1,53 @@
 //SPDX-License-Identifier: Unlicense
-pragma solidity 0.8.6;
 
-import "./@openzeppelin/contracts/token/ERC20/ERC20.sol";
+pragma solidity 0.8.6;
 
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 /**
  * @title ERC20TimedMint
- * @author Javier Gonzalez
+ * @author Javier Gonzalez and Marco Huberts
  * @dev Implementation of minting functionality for a mintable token.
  * @notice ERC20TimedMint inherits the ERC20 functionality and prevents 
- *         minting within a timeframe.
- */
+  *         minting within a timeframe.
+  */
 contract ERC20TimedMint is ERC20 {
-    uint256 timeUntilNexMint
 
-    constructor(
+    uint256 public timeUntilNextMint;
+    uint256 private mintCap;
+
+    event Claim(address claimant, uint256 amount);
+    
+    constructor (
         string memory name,
         string memory symbol,
+        uint256 _mintCap,
         uint256 _timeDelay
-    ) ERC20(name, symbol) {
-        timeUntilNexMint = block.timeStamp + _timeDelay
+    )   
+        ERC20(name, symbol) 
+    {
+        timeUntilNextMint = block.timestamp + _timeDelay;
+        mintCap = _mintCap;
+    }
+
+    function claimTokens (
+        uint256 amount, 
+        address _receiver
+    ) 
+        external 
+    {
+        require(_receiver == msg.sender); 
+        _transfer(address(this), msg.sender, amount);
+        emit Claim(msg.sender, amount);
+    }
+
+    function _mint(address to, uint256 amount)
+        internal
+        override(ERC20)
+    {
+        require(block.timestamp >= timeUntilNextMint, "ERC20: Cannot mint yet");
+        require(amount <= mintCap, "ERC20: Mint exceeds maximum amount");
```

Comment:
I would like there to be the option of setting a new timestamp at this moment. Otherwise, you could call the new _mint function whenever after the initial time delay. 
I would like to have the option of calling a function to set new time delay internally as well. 
So ideally, there would be another `internal` function maybe `_setTimeUntilNextMint` that sets the _timeDelay from the constructor (meaning we should store that).